### PR TITLE
refactored not logged in state to smaller parts

### PR DESCRIPTION
**Pull request by jmatosp**
A short summary.

Important changes:
- [x] Something worth noting for reviewers.

Mandatory:
- [x] Added documentation.

Closes #

### CODE REVIEW

File: client/knode/validator/states.go [hunk 1]
```diff
@@ -75,10 +41,53 @@ func (val *validator) notLoggedInState() stateFn {
...
+		return val.startValidating
 	}
 
+	return val.notLoggedInState
+}
+
+func (val *validator) notLoggedInState() stateFn {
+	chainHeadCh := make(chan core.ChainHeadEvent)
+	chainHeadSub := val.chain.SubscribeChainHeadEvent(chainHeadCh)
+	defer chainHeadSub.Unsubscribe()
```
Comment (jmatosp) [inline client/knode/validator/states.go]: @rgeraldes can you explain what this is doing?  this func only has a call so the defer unsubscribes right after

Comment (rgeraldes) [reply client/knode/validator/states.go]: chainHeadSub lets us know when there's a new chain head (block) > check the state again to verify if the user is a validator.

Comment (jmatosp) [reply client/knode/validator/states.go]: but chainHeadSub is not in use anywhere. And it's closes the channel as soon at exits that func

Comment (rgeraldes) [reply client/knode/validator/states.go]: that's true - we can remove it for now.

**Comment by jmatosp** [comment]
@rgeraldes can you re-review

File: client/knode/validator/states.go [hunk 2]
```diff
@@ -29,44 +29,10 @@ type work struct {
...
-				log.Crit("Failed to verify the voter registration", "err", err)
-			}
-			if receipt.Status == types.ReceiptStatusFailed {
-				log.Crit("Failed to register the validator - receipt status failed")
-			}
-		}
-
-	} else {
+	if val.isBlockZero() {
 		isVoter, err := val.consensus.IsValidator(val.walletAccount.Account().Address)
```
Comment (rgeraldes) [inline client/knode/validator/states.go]: This check is not necessary as according to the contract, in order to be a genesis on block 0, the user needs to be registered as a validator. (Genesis flag is set during the registration process, there's no way around it).

**Review by rgeraldes — APPROVED** [review]
Left a small comment about a possible optimisation.

File: client/knode/validator/states.go (hunk 2 above)
Comment (jmatosp) [reply client/knode/validator/states.go]: not sure what you mean with remove the check.

Does that means that a genesis validator never does a deposit?

Comment (jmatosp) [reply client/knode/validator/states.go]: Oh I see, got the line worng, so as long is a genesis and block zero can just start validating imediatly

Comment (rgeraldes) [reply client/knode/validator/states.go]: yes

**Pull request merged**